### PR TITLE
Add WebSub Spec

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#2162](https://github.com/ballerina-platform/ballerina-standard-library/issues/2161)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    * 2.1. [Listener](#21-listener)
+      * 2.1.1. [Configuration](#211-configuration)
+      * 2.1.2. [Initialization](#212-initialization)
+      * 2.1.3. [Methods](#213-methods)
+    * 2.2 [Subscriber Service](#22-subscriber-service)
+      * 2.2.1. [Methods](#221-methods)
+        * 2.2.1.1. [onSubscriptionValidationDenied](#2211-onsubscriptionvalidationdenied)
+        * 2.2.1.2. [onSubscriptionVerification](#2212-onsubscriptionverification)
+        * 2.2.1.3. [onUnsubscriptionVerification](#2213-onunsubscriptionverification)
+        * 2.2.1.1. [onEventNotification](#2214-oneventnotification)
```
Comment (chamil321) [inline docs/spec/spec.md:27]:
```suggestion
        * 2.2.1.4. [onEventNotification](#2214-oneventnotification)
```

File: docs/spec/spec.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+public isolated function init(int|http:Listener listenTo, *ListenerConfiguration config) returns websub:Error? 
+```
+
+#### 2.1.3. Methods
+
+Following APIs should be available in the `websub:Listener` to dynamically attach `websub:SubscriberService` objects to 
+it.
+```ballerina
+# Attaches the provided `websub:SubscriberService` to the `websub:Listener`.
+# ```ballerina
```
Comment (chamil321) [inline docs/spec/spec.md:104]:
```suggestion
```

File: docs/spec/spec.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+# check websubListenerEp.attach('service, "/subscriber");
+# ```
```
Comment (chamil321) [inline docs/spec/spec.md:106]:
```suggestion
```

Comment (chamil321) [reply docs/spec/spec.md:106]: Let's remove these inner quotes. Due to the out quotes, inner quotes will not function I think. Check the other places as well

File: docs/spec/spec.md [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+
+As per the WebSub Specification a `subscriber` should be able to subscribe to a particular `topic` in a specific `hub`. 
+As the counterpart subscriber should be able to unsubscribe from a previously subscribed `topic` in a specific `hub`. 
+Even though the specification does not clearly define when this unsubscription should happen, it is obvious that this 
+should happen whenever `subscriber` is terminated.  
+
+The key functionalities expected from the unsubscription flow as follows:  
+- Developer should manually enable this feature by configuring `unsubscribeOnShutdown` in `websub:SubscriberServiceConfig` annotation.
+- Developer should be able to configure unsubscription verification time-out using `gracefulShutdownPeriod` configuration in `websub:ListenerConfiguration`.
+- Unsubscription flow should initiate whenever graceful stop is invoked in `websub:Listener`.
```
Comment (chamil321) [inline docs/spec/spec.md:328]:
```suggestion
- Unsubscription flow should be initiated whenever the graceful stop is invoked in `websub:Listener`.
```

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**